hummingbot/client/hummingbot_application.py

```diff
@@ -218,125 +210,32 @@ def _initialize_markets(self, market_names: List[Tuple[str, List[str]]]):
+                    connector = create_paper_trade_market(market_name, trading_pairs)
                 except Exception:
                     raise
                 paper_trade_account_balance = global_config_map.get("paper_trade_account_balance").value
                 for asset, balance in paper_trade_account_balance.items():
-                    market.set_balance(asset, balance)
-
-            elif market_name == "binance":
-                binance_api_key = global_config_map.get("binance_api_key").value
-                binance_api_secret = global_config_map.get("binance_api_secret").value
-                market = BinanceMarket(
-                    binance_api_key,
-                    binance_api_secret,
-                    trading_pairs=trading_pairs,
-                    trading_required=self._trading_required
-                )
+                    connector.set_balance(asset, balance)
 
-            elif market_name == "radar_relay":
-                assert self.wallet is not None
-                market = RadarRelayMarket(
-                    wallet=self.wallet,
-                    ethereum_rpc_url=ethereum_rpc_url,
-                    trading_pairs=trading_pairs,
-                    trading_required=self._trading_required,
-                )
+            elif connector_name in EXCHANGES or connector_name in DERIVATIVES:
+                keys = dict((key, value.value) for key, value in dict(filter(lambda item: connector_name in item[0], global_config_map.items())).items())
+                connector_class = get_connector_class(connector_name)
+                connector = connector_class(**keys, trading_pairs=trading_pairs, trading_required=self._trading_required)
 
-            elif market_name == "bamboo_relay":
+            elif connector_name in DEXES:
                 assert self.wallet is not None
-                use_coordinator = global_config_map.get("bamboo_relay_use_coordinator").value
-                pre_emptive_soft_cancels = global_config_map.get("bamboo_relay_pre_emptive_soft_cancels").value
-                market = BambooRelayMarket(
-                    wallet=self.wallet,
-                    ethereum_rpc_url=ethereum_rpc_url,
-                    trading_pairs=trading_pairs,
-                    use_coordinator=use_coordinator,
-                    pre_emptive_soft_cancels=pre_emptive_soft_cancels,
-                    trading_required=self._trading_required,
-                )
+                keys = dict((key, value.value) for key, value in dict(filter(lambda item: connector_name in item[0], global_config_map.items())).items())
+                connector_class = get_connector_class(connector_name)
+                connector = connector_class(**keys, wallet=self.wallet, ethereum_rpc_url=ethereum_rpc_url, trading_pairs=trading_pairs, trading_required=self._trading_required)
+                # TO-DO for DEXes: rename all extra argument to match key in global_config_map
```

Comment:
Is this still need to be done?

Reply:
No, it's done.
I'd remove the comment now. 
Thanks.